dart/tool/dart/generate_dart.dart

```diff
@@ -1113,10 +1125,23 @@ class MethodParser extends Parser {
     while (peek().text != ')') {
       Token type = expectName();
       TypeRef ref = new TypeRef(_coerceRefType(type.text));
-      while (consume('[')) {
-        expect(']');
-        ref.arrayDepth++;
+      if (peek().text == '[') {
+        while (consume('[')) {
+          expect(']');
+          ref.arrayDepth++;
+        }
+      } else if (peek().text == '<') {
+        // handle generics
+        expect('<');
+        ref.genericTypes = [];
+        while (peek().text != '>') {
+          Token genericTypeName = expectName();
+          ref.genericTypes.add(new TypeRef(_coerceRefType(genericTypeName.text)));
+          consume(',');
```

Comment:
I think the commas are optional here, in that `a, b, c` - c won't have a trailing comma.